src/renderer/redux/sagas/i18n.ts

```diff
@@ -6,16 +6,20 @@
 // ==LICENSE-END==
 
 import { i18nActions } from "readium-desktop/common/redux/actions";
-import { takeTyped } from "readium-desktop/common/redux/typed-saga";
 import { diRendererGet } from "readium-desktop/renderer/di";
-import { all, call } from "redux-saga/effects";
+import { all, call, takeEvery } from "redux-saga/effects";
 
-function* localeWatcher() {
-    while (true) {
-        const action = yield* takeTyped(i18nActions.setLocale.build);
-        const translator = diRendererGet("translator");
+function* setLocale(action: i18nActions.setLocale.TAction) {
+    const translator = diRendererGet("translator");
+
+    const translatorSetLocale = () =>
         translator.setLocale(action.payload.locale);
-    }
+
+    yield call(translatorSetLocale);
```

Comment:
Same remark here about `await` on the `setLocale()` Promise-returning function.

Reply:
https://github.com/readium/readium-desktop/pull/907/commits/73dbaf736242aa558ac43cc7b35d05960b1c9af6